--- a/arch/arm/boot/dts/Makefile	2022-02-14 09:13:30.000000000 +1300
+++ b/arch/arm/boot/dts/Makefile	2022-02-19 14:55:05.334709503 +1300
@@ -271,7 +271,8 @@
 	intel-ixp43x-gateworks-gw2358.dtb \
 	intel-ixp42x-netgear-wg302v2.dtb \
 	intel-ixp42x-arcom-vulcan.dtb \
 	intel-ixp42x-gateway-7001.dtb \
-	intel-ixp42x-usrobotics-usr8200.dtb
+	intel-ixp42x-usrobotics-usr8200.dtb \
+	intel-ixp43x-watchguard-xtm21w.dtb
 dtb-$(CONFIG_ARCH_KEYSTONE) += \
 	keystone-k2hk-evm.dtb \
 	keystone-k2l-evm.dtb \
